src/Storages/RocksDB/StorageEmbeddedRocksDB.cpp

```diff
@@ -519,11 +536,24 @@ SinkToStoragePtr StorageEmbeddedRocksDB::write(
 static StoragePtr create(const StorageFactory::Arguments & args)
 {
     // TODO custom RocksDBSettings, table function
-    if (!args.engine_args.empty())
+    auto engine_args = args.engine_args;
+    if (engine_args.size() > 2)
         throw Exception(
-            "Engine " + args.engine_name + " doesn't support any arguments (" + toString(args.engine_args.size()) + " given)",
+            "Engine " + args.engine_name + " requires less than 2 parameters (" + toString(engine_args.size()) + " given)",
```

Comment:
```suggestion
            "Engine " + args.engine_name + " requires at most 2 parameters (" + toString(engine_args.size()) + " given)",
```
You can also use fmt style strings creation:
```
throw Exception(ErrorCodes::NUMBEROF_ARGUMENTS_DOESNT_MATCH, "Engine {} requires at most 2 parameters. ({} given). Correct usage: EmbeddedRocksDB([ttl, read_only])", args.engine_name, engine_args.size());
```